R/prep_scores.R

```diff
@@ -1,7 +1,87 @@
-prep_scores <- function(results_portfolio = NULL, peers_results_aggregated = NULL,
-                        asset_class = "equity") {
-  if (is.null(results_portfolio)) {
-    data_out <- use_toy_data("scores") %>% filter(asset_class == .env$asset_class)
+#' Prepare data input for plotting aggregate climate scores
+#'
+#' @description Prepare data input for plotting aggregate climate scores based
+#' on PACTA for investors output files. These files must have been wrangled with
+#' `prep_data_executive_summary()` before they can be passed to this function.
+#'
+#' @param results_portfolio Data frame that contains pre-wrangled portfolio
+#'   level PACTA results from a PACTA for investors analysis.
+#' @param peers_results_aggregated Data frame that contains pre-wrangled
+#'   aggregate peer group level PACTA results from a PACTA for investors
+#'   analysis.
+#' @param asset_class Character vector of length 1. Must be either `equity` or
+#'   `bonds`.
+#' @param scenario_source Character vector of length 1. Must be a
+#'   `scenario_source` featured in the `scenario_thresholds` data set.
+#'
+#' @return data.frame
+prep_scores <- function(results_portfolio,
+                        peers_results_aggregated,
+                        asset_class,
+                        scenario_source = "GECO2021") {
+
+  # validate inputs
+  if (!asset_class %in% c("bonds", "equity")) {
```

Comment:
nit: To keep the code cleaner and easier to read I'd move all the data checks to a wrapper function `check_data_prep_scores()` or something like that.

Reply:
valid. but do you not think this would contradict the comment about keeping the R/ folder clean a bit? Alternatively, we could add one file R/check_data.R that holds all such checks, should there be more for the remaining prep functions?

Reply:
Done. Also moved all other directly related functions into prep_scores.R, which keeps the directory clean